lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,11 +20,13 @@ def as_json
 
         def type_for(serializer)
           return serializer._type if serializer._type
-          if ActiveModelSerializers.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
-          else
-            serializer.object.class.model_name.plural
+          type = serializer.object.class.to_s.split('::')
+          type.map! { |t| t.underscore.downcase }
```

Comment:
It makes perfect sense to use KeyTransform (with according setting), but what if someone needs to keys underscored and type dasherized? Is it unlikely? Or we're gonna use methods from KeyTransform with another setting?